Makefile

```diff
@@ -1,29 +1,61 @@
-CC      = gcc
-CFLAGS  = -c -g -O2 -Wall
-LDFLAGS =
-
-SRC     = $(sort $(wildcard *.c))
-OBJ     = $(SRC:.c=.o)
+CC           = gcc
+CFLAGS       = -g -O2 -Wall
+LDFLAGS      = -L. -lmediacheck
+SHARED_FLAGS = -fPIC -fvisibility=hidden
 
+ARCH    := $(shell uname -m)
 GIT2LOG := $(shell if [ -x ./git2log ] ; then echo ./git2log --update ; else echo true ; fi)
 GITDEPS := $(shell [ -d .git ] && echo .git/HEAD .git/refs/heads .git/refs/tags)
 VERSION := $(shell $(GIT2LOG) --version VERSION ; cat VERSION)
 BRANCH  := $(shell [ -d .git ] && git branch | perl -ne 'print $$_ if s/^\*\s*//')
 PREFIX  := checkmedia-$(VERSION)
 
-%.o: %.c
-	$(CC) $(CFLAGS) -DVERSION=\"$(VERSION)\" -o $@ $<
+MAJOR_VERSION := $(shell $(GIT2LOG) --version VERSION ; cut -d . -f 1 VERSION)
+
+LIB_NAME     = libmediacheck
+LIB_FILENAME = $(LIB_NAME).so.$(VERSION)
+LIB_SONAME   = $(LIB_NAME).so.$(MAJOR_VERSION)
+
+DIGEST_SRC  = $(wildcard md5.c sha*.c)
+DIGEST_OBJ  = $(DIGEST_SRC:.c=.o)
+
+ifneq ($(filter x86_64, $(ARCH)),)
+LIBDIR  = /usr/lib64
+else
+LIBDIR  = /usr/lib
+endif
+
+all: checkmedia digestdemo
 
-all: checkmedia
+checkmedia: checkmedia.c $(LIB_FILENAME) mediacheck.h
```

Comment:
is here mediacheck.h needed? I expect that header change cause `$(LIB_FILENAME)` rebuild resulting also in this rebuild, not? and same below for digestdemo.